### DISCUSSION TITLE
Streaming for multipart/form-data

### DISCUSSION THREAD

**Discussion by ymmt2005** [ogen-go/ogen#1115] [Ideas]
The current ogen generated code uses [*Request.ParseMultipartForm](https://pkg.go.dev/net/http#Request.ParseMultipartForm), which stores data in the filesystem.
As we sometimes do not want to store data in the filesystem, it would be great if ogen
could generate code using [*Request.MultipartReader](https://pkg.go.dev/net/http#Request.MultipartReader) which can process the request body as a stream.

**Comment by ymmt2005** [comment]
This would generate a fundamentally different interface from the non-streaming version.
I think we should add an extension `x-ogen-multipart-streaming`, just like we do for JSON streaming.

```yaml
    post:
      requestBody:
        content:
          multipart/form-data:
            x-ogen-multipart-streaming: true
```

**Reply by k1ng440** [reply]
This would be great.
